price-discovery/src/redeem_token.rs

```diff
@@ -0,0 +1,123 @@
+elrond_wasm::imports!();
+use hex_literal::hex;
+
+pub const LAUNCHED_TOKEN_REDEEM_NONCE: u64 = 1;
+pub const ACCEPTED_TOKEN_REDEEM_NONCE: u64 = 2;
+
+const SFT_TOKEN_TYPE_NAME: &[u8] = b"SFT";
```

Comment:
META type has to be used. this is a financial token

Reply:
Done.